feat: document new React usage (#1791)
getsentry/sentry-docs: src/collections/_documentation/platforms/javascript/react.md

keywords: ["reactjs"]
---
<!-- WIZARD -->
To use Sentry with your React application, you will need to use `@sentry/react` (Sentry’s Browser React SDK).

{% capture __alert_content -%}

> `@sentry/react` is a is a wrapper around the `@sentry/browser` package, with added functionality related to React. All methods available in the `@sentry/browser` package also can also be imported from `@sentry/react`.

Reviewer: ```suggestion
`@sentry/react` is a wrapper around the `@sentry/browser` package, with added functionality related to React. All methods available in the `@sentry/browser` package can also be imported from `@sentry/react`.
```

{%- endcapture -%}
{%- include components/alert.html
  title="Note"
  content=__alert_content
  level="info"
%}

{% include_relative getting-started-install/react.md %}

## Connecting the SDK to Sentry

After you've completed setting up a project in Sentry, Sentry will give you a value which we call a _DSN_ or _Data Source Name_. It looks a lot like a standard URL, but it’s just a representation of the configuration required by the Sentry SDKs. It consists of a few pieces, including the protocol, public key, the server address, and the project identifier.

{% include_relative getting-started-dsn/react.md %}

> On its own, `@sentry/react` will report any uncaught exceptions triggered from your application.

Reviewer: ```suggestion
On its own, `@sentry/react` will report any uncaught exceptions triggered by your application.
```

```
<!-- ENDWIZARD -->

## Error Boundaries

> If you’re using React 16 or above, Error Boundaries are an important tool for defining the behavior of your application in the face of errors. The `@sentry/react` package exposes an error boundary component that automatically sends JavaScript errors from inside a React component tree to Sentry.

Reviewer: ```suggestion
If you're using React 16 or above, Error Boundaries are an essential tool for defining your application's behavior in the face of errors. The `@sentry/react` package exposes an error boundary component that automatically sends JavaScript errors from inside a React component tree to Sentry.
```

  level="warning"
%}

> In the example below, when the `<Example />` component hits an error, the `<Sentry.ErrorBoundary>` component will send data about that error and the component tree to Sentry, and open a user feedback dialog, and render a fallback UI.

Reviewer: ```suggestion
In the example below, when the `<Example />` component hits an error, the `<Sentry.ErrorBoundary>` component will send data about that error and the component tree to Sentry, open a user feedback dialog, and render a fallback UI.
```

```jsx
import React from 'react';
import * as Sentry from '@sentry/react';

import {Example} from '../example';

function FallbackComponent() {
  return (
    <div>An error has occured</div>
  )
}

class App extends React.Component {
  render() {
    return (
      <Sentry.ErrorBoundary fallback={FallbackComponent} showDialog>
        <Example />
      </Sentry.ErrorBoundary>
    )
  }
}

export default App;
```

### ErrorBoundary Options

> The ErrorBoundary component exposes a variety props that can be passed in for extra configuration. There are no required options, but it is highly recommended that a fallback component be set.

Reviewer: ```suggestion
The ErrorBoundary component exposes a variety of props that can be passed in for extra configuration. There are no required options, but we highly recommend that you set a fallback component.
```

#### `showDialog` (boolean)

If a [Sentry User Feedback Widget]({%- link _documentation/enriching-error-data/user-feedback.md -%}?platform={{ include.platform }}) should be rendered when the Error Boundary catches an error.

#### `dialogOptions` (Object)

Options that are passed into the Sentry User Feedback Widget. See all possible customization options [here]({%- link _documentation/enriching-error-data/user-feedback.md -%}?platform={{ include.platform }}#customizing-the-widget)

#### `fallback` (React.ReactNode or Function)

> A fallback component that gets rendered when the error boundary encounters an error. Can either provide a React Component, or a function that returns React Component as a valid fallback prop. If a function is provided, the function will be called with the error and the component stack at time of error, and a function that resets the error boundary to it's regular state.

Reviewer: ```suggestion
A fallback component that gets rendered when the error boundary encounters an error. It can either provide a React Component, or a function that returns React Component as a valid fallback prop. If you provide a function, Sentry will call the function with the error and the component stack at the time of the error. 
```
The original last sentence was difficult to parse. I broke up the sentence and deleted second half of the sentence because I couldn't understand what it was trying to communicate. Feel free to rephrase what I wrote.

Author: Yes it was, good point. I reworded based on your suggestion. Thanks!

#### `onError` (Function)

> A function that gets called when the Error Boundary encounter an error. `onError` is useful if you want to propogate the error into a state management library like Redux, or if you want to check any side effects that could have occured due to the error.

Reviewer: ```suggestion
A function that gets called when the Error Boundary encounters an error. `onError` is useful if you want to propagate the error into a state management library like Redux, or if you want to check any side effects that could have occurred due to the error.
```

#### `onMount` (Function)

A function that gets called on ErrorBoundary `componentDidMount()`

#### `onUnmount` (Function)

A function that gets called on ErrorBoundary `componentWillUnmount()`

### ErrorBoundary Usage

#### Setting a Fallback Function (Render Props)

> Below is an example where a fallback prop, using the [render props approach](https://reactjs.org/docs/render-props.html), is used to display a fallback UI on error, and gracefully return to a normal component state when reset.

Reviewer: ```suggestion
Below is an example where a fallback prop, using the [render props approach](https://reactjs.org/docs/render-props.html), is used to display a fallback UI on error, and gracefully return to a standard component state when reset.
```

We try to avoid the word "normal" because it's not inclusive language.

Author: Ok will keep that in mind 👍
